proxy/envoy/config.go

```diff
@@ -411,6 +413,37 @@ func buildRoutes(svc *model.Service, rulesMap map[string][]*config.RouteRule,
 	return routes
 }
 
+func buildRouteHeaders(headerMatches map[string]*config.StringMatch) []Header {
```

Comment:
Rename to buildHeaderMatches and the other one to buildHeaderMatch as it can be reused in fault filters as well.